### ISSUE TITLE
Faild pulling proxy image - error during switch hub

### ISSUE THREAD

**Issue by lukaszpy** [istio/istio#29243]
I have faced problem with pulling proxy image form dockerhub. 

`Failed to pull image "docker.io/istio/proxyv2:1.8.0": rpc error: code = Unknown desc = Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit`

I want to switch to my private registry:
`istioctl install --set hub=dockerhub.bitit.net:5555 --set values.global.imagePullSecrets=gitlab-registry-secret-bitit`

But I'm getting error:

```
This will install the Istio profile into the cluster. Proceed? (y/N) y
Detected that your cluster does not support third party JWT authentication. Falling back to less secure first party JWT. See https://istio.io/v1.8/docs/ops/best-practices/security/#configure-third-party-service-account-tokens for details.
Run the command with the --force flag if you want to ignore the validation error and proceed.
Error: failed to install manifests: json: cannot unmarshal string into Go value of type []json.RawMessage
```

[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
want to use private registry with private access data (imagePullSecrets)

**Steps to reproduce the bug**
execute
`istioctl install --set hub=dockerhub.bitit.net:5555 --set values.global.imagePullSecrets=gitlab-registry-secret-bitit`
on cluster with running istio.

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**
client version: 1.8.0
control plane version: 1.8.0
data plane version: 1.8.0 (41 proxies)

Client Version: v1.17.8
Server Version: v1.17.8

**How was Istio installed?**
istioctl install

**Environment where bug was observed (cloud vendor, OS, etc)**
private cluster

**Comment by howardjohn** [comment]
image pull secrets is a list but you are passing a single value

**Comment by lukaszpy** [comment]
is it possible to pass there list ?

**Comment by howardjohn** [comment]
https://istio.io/latest/docs/setup/install/istioctl/#list-item-path-selection

I suggest using the yaml input though

**Comment by lukaszpy** [comment]
But is it possible to generate manifest from current conf, modify and apply it again or should I  purge istio flom cluster ?

**Comment by morvencao** [comment]
Yes, you can generate the manifests with 
```
istioctl manifest generate -f <iop_conf.yaml>
```

to generate the manifests before you apply it to the cluster.

Also, if you just want to override the hub to private registry and add image pull secret, then you can apply it again.

**Comment by lukaszpy** [comment]
@morvencao it works for me partialy...

But finally I have solved my problems by:
1. generate manifest: istioctl manifest generate > generated-manifest.yaml
2. create yaml file with only ConfigMap :
```
 name: istio-sidecar-injector
 namespace: istio-system
```
3. and I have apllyied it. 
But it gives me error:
`MountVolume.SetUp failed for volume "istio-token" : failed to fetch token: the API server does not have TokenRequest endpoints enabled`
I have read: https://istio.io/latest/docs/ops/best-practices/security/#configure-third-party-service-account-tokens
I I modifed config map with:
```
        "hub": "dockerhub.bitit.net:5555/istio",
        "imagePullPolicy": "",
        "imagePullSecrets": ["gitlab-registry-secret-bitit"],
        "istioNamespace": "istio-system",
        "istiod": {
          "enableAnalysis": false
        },
        "jwtPolicy": "first-party-jwt",
```
the problem was in **"jwtPolicy": "first-party-jwt"**

Now I downloading proxy image form private registry.
